docs/guides/input.md

```diff
@@ -0,0 +1,292 @@
+# Input
+
+_Since version 4.27.0_
+
+[:rocket: Quick Reference](#quick-reference)
+
+When you're building software, it's common to split your code up into small pieces. These might be functions in a node process, or framework components. These small pieces often need to declare **dependencies** - bits of information/code that they're not responsible for, but that they rely on to serve their purpose.
+
+The best way to declare dependencies to an XState machine is called `input`. You can specify input like so:
+
+```ts
+const machine = createMachine(
+  {
+    entry: (context, event, meta) => {
+      console.log(meta.state.input.lyric); // 'In west Philadelphia, born and raised'
+    }
+  },
+  {
+    input: {
+      lyric: 'In west Philadelphia, born and raised'
+    }
+  }
+);
+```
+
+Or, by using the `.withConfig` method:
+
+```ts
+const machineWithDifferentInput = machine.withConfig({
+  input: {
+    lyric: 'In the playground, where I spend most of my days'
+  }
+});
+```
+
+The input is then available in any action/service/guard, in the `meta` argument.
+
+```ts
+const machine = createMachine(
+  {
+    entry: [
+      (context, event, meta) => {
+        console.log(meta.state.input.foo);
+      }
+    ],
+    invoke: {
+      src: (context, event, meta) => {
+        console.log(meta.state.input.foo);
+      },
```

Comment:
Since `meta.state` is currently not available to invoke creators this should be removed from an example and from other places mentioning this.

Reply:
Isn't it? Ah, crap

Reply:
That is not hard to fix, but I... don't want to 😅 this is pending on the outcome of this: https://github.com/statelyai/xstate/pull/2779#discussion_r749321830 . I think we should decide on this asap - as it would be bad to introduce a new thing into a potentially deprecated "argument". If we can spare people from refactoring this in the future this would be awesome.